### PR TITLE
2.0.4 improvments

### PR DESCRIPTION
**Pull request by yarikbratashchuk**

### CODE REVIEW

File: lib/img.js [hunk 1]
```diff
@@ -16,6 +21,11 @@ Blinkloader.getRelevantWidth = function(width) {
 };
 
 Blinkloader.hash = function(s) {
+  if (typeof s !== 'string') {
+    Blinkloader.err(Blinkloader.badArgErr('hash', arguments));
+    return 0;
+  }
```
Comment (mac-r) [inline lib/img.js]: maybe we could return `null` and add related validations (`not null`), where this func is invoked?

File: lib/img.js [hunk 2]
```diff
@@ -24,18 +34,28 @@ Blinkloader.hash = function(s) {
...
+  if (parser.hostname === window.location.hostname) {
     return true;
   }
   return false;
 };
 
 Blinkloader.trimHost = function(link) {
+  if (typeof link !== 'string') {
+    Blinkloader.err(Blinkloader.badArgErr('trimHost', arguments));
+    return link;
```
Comment (mac-r) [inline lib/img.js]: same here. maybe we could return null and add related validations (not null), where this func is invoked?
returning `link`, which is not string doesn't make sense here.

File: lib/1_window.js [hunk 3]
```diff
@@ -82,6 +87,11 @@ Blinkloader.retinaSupport = function() {
 }
 
 Blinkloader.webpSupport = function(callback) {
+  if (typeof callback !== 'function') {
+    Blinkloader.err(Blinkloader.badArgErr('webpSupport', arguments));
+    return;
```
Comment (mac-r) [inline lib/1_window.js]: why are you returning nothing here, but in a similar case (ex: `Blinkloader.prepareCdnImageUrl`) you return a callback with `null`? it's more consistent to return `callback(null)` here and to add `not null` validations, where this function is invoked.

File: lib/img.js [hunk 4]
```diff
@@ -46,6 +66,11 @@ Blinkloader.trimHost = function(link) {
 };
 
 Blinkloader.getImageName = function(imageUrl) {
+  if (typeof imageUrl !== 'string') {
+    Blinkloader.err(Blinkloader.badArgErr('getImageName', arguments));
+    return '';
```
Comment (mac-r) [inline lib/img.js]: how about returning `null` here as well and then validate the value where it's invoked?

File: lib/1_window.js (hunk 3 above)
Comment (yarikbratashchuk) [reply lib/1_window.js]: we can't invoke callback(null) if callback is not a function

File: lib/img.js (hunk 1 above)
Comment (yarikbratashchuk) [reply lib/img.js]: the idea of returning `0` here is senseless

Comment (yarikbratashchuk) [reply lib/img.js]: i think returning null is better (makes sense at least)

File: lib/img.js (hunk 2 above)
Comment (yarikbratashchuk) [reply lib/img.js]: agreed

File: lib/img.js (hunk 4 above)
Comment (yarikbratashchuk) [reply lib/img.js]: agreed

File: lib/1_window.js (hunk 3 above)
Comment (mac-r) [reply lib/1_window.js]: ohh, yeah, correct 👍

**Pull request merged**